crates/ruff_python_parser/src/parser/expression.rs

```diff
@@ -2777,18 +2777,24 @@ impl<'src> Parser<'src> {
             unreachable!()
         };
 
-        if !matches!(kind, IpyEscapeKind::Magic | IpyEscapeKind::Shell) {
-            // This should never occur as the lexer won't allow it.
```

Comment:
Can you explain to me why the lexer emits the `Magic` and `Shell` tokens contrary to what the comment says? Should we fix the lexer instead?

Reply:
That makes more sense, yeah. I'm pushing a change to have it fixed at the lexer level by adding an `in_expression_context` parameter to `lex_ipython_escape_command` and preventing Help kind conversion when `true` (e.g., `%foo?` stays Magic in expression contexts).

I think it makes sense to keep the parser check as a defensive measure for edge cases like `with a,?b`, where Help tokens can still be lexed at the start of logical lines. Does this approach look right?